src/ts-typings/api.d.ts

```diff
@@ -58,15 +58,13 @@ type AttributeReflectionConfig = AttributeReflectionBaseType | {
   source?: AttributeReflectionBaseType,
   target?: AttributeReflectionBaseType
 }
-export interface PropOptions<El, T> {
+export interface PropOptions {
   attribute?: AttributeReflectionConfig;
-  coerce?: (value: any) => T | null | undefined;
-  default?: T | null | undefined | ((elem: El, data: { name: string; }) => T | null | undefined);
-  deserialize?: (value: string | null) => T | null | undefined;
-  get?: <R>(elem: El, data: { name: string; internalValue: T; }) => R;
-  initial?: T | null | undefined | ((elem: El, data: { name: string; }) => T | null | undefined);
-  serialize?: (value: T | null | undefined) => string | null;
-  set?: (elem: El, data: { name: string; newValue: T | null | undefined; oldValue: T | null | undefined; }) => void;
+  coerce?: <T>(value: any) => T | null | undefined;
+  default?: any | ((elem: HTMLElement, data: { name: string; }) => any);
```

Comment:
Yeah I know. It's fine for now, we are using decorators for defining props, which feels more first class citizens to TS users ( because definition and type annotation of prop are together )